### ISSUE TITLE
Licensing :(

### ISSUE THREAD

**Issue by jokeyrhyme** [SavedInstances/SavedInstances#54]
Hi, I'm the original author of this addon, from it's initial release up to 4.2.4 when oscarucb graciously took over (I think, I could have the details wrong)

Firstly, I'm glad to see others still find SavedInstances useful, and it's terrific that others have taken its stewardship upon themselves

I was contacted about the licensing issues with the current last-published version on Curse.com, and am participating in a discussion over there:
https://www.wowace.com/projects/saved_instances/issues/295

> I'm not sure this is strictly legal to use any of the code that was licensed as All Rights Reserved
> 
> This is the commit where oscarucb changed the license from BSD to All Rights Reserved: https://github.com/eTzmNcbkrng/SavedInstances/commit/c85c585749912e832c0223276d32fc59314266aa
> This was then tagged as version 5.7.3
>
> So, all the code up to this commit was BSD and able to be used as anyone sees fit
> Everything from this commit onwards is completely locked up and can only be published / modified / relicensed / etc by oscarucb themselves

I think what you are doing is amazing, and I want to find a way to grant you the proper access required to publish this to Curse.com officially
But, I feel like we might be required to drop all the commits that were marked as All Rights Reserved

I'm no copyright lawyer, and I want to have an open dialogue and see what other people think
I could also just be a crazy old programmer with silly ideas :P

Anyone else have any thoughts on how to proceed here?

**Comment by TruthNZ** [comment]
A removal of the BSD licence from the code would illegal.
New code added by someone else can be under a different licence - but the original code must still include the BSD licence (and you must specify which parts of code come under which licence). Thus the commit that removed the reference would be illegal.

That doesn't necessarily stop the code that oscarucb added being still licenced under his choice of licence - but since we could claim it's based on BSD licence code ... let's just say it could get a bit messy, and we'd probably need proper legal advice.

**Comment by Stanzilla** [comment]
I knew right from the start when I made my personal fork that his problem would come up some day, I thought oscarub would come back and was prepared to just offer PRs instead of keeping my fork. Since that has not happened (yet?) I would suggest trying to contact him and ask what his stance is, I never tried it and I do not know if anyone else ever did.

**Comment by TruthNZ** [comment]
If we can get his permission to continue - that would solve any problems.
And ideally get him to add someone else on the curse site to allow us to update that too.

**Comment by jokeyrhyme** [comment]
Hmmm, https://github.com/oscarucb ? @oscarucb is that you?

**Comment by gOOvER** [comment]
Maybe Torhal can help:

https://github.com/Torhal

**Comment by jokeyrhyme** [comment]
I think, at best, when the license was changed to All Rights Reserved, it only affected new code from that point on with all prior code still being available under BSD

It's perfectly legal and normal to change the license of a project over time, it just doesn't impact previous versions of the code

I'd love to hear from oscarucb, they did some great work over the years

**Comment by gOOvER** [comment]
answer from Torhal:

"	
Torhal Today at 08:43
Hey there,
This is something the original author would need to act upon. Otherwise, feel free to create a forked project from the pre-ARR change and also change the project name.
Take care,
Torhal"

**Comment by jokeyrhyme** [comment]
I've submitted a support request to see if I can gain administrative permissions for the project

**Comment by jokeyrhyme** [comment]
@torhal has restored my admin access to the project :)

I will leave @oscarucb with membership for now, in case they wish to continue their good work sometime in future

If we take a look at the contribution graph for this fork: https://github.com/eTzmNcbkrng/SavedInstances/graphs/contributors

It would seem that @Stanzilla and @eTzmNcbkrng are good candidates for additional authors / admins
If interested parties (even beyond these fine folks) wish to have author (or other) permissions on the project, please share your Curse account name and the specific role or permissions you'd like

These are the default roles:
![image](https://user-images.githubusercontent.com/479816/31312920-5c5c22b8-ac1d-11e7-8daf-f0362d625c53.png)

And these are the individual permissions that I'm guessing are split logically across those roles:
![image](https://user-images.githubusercontent.com/479816/31312912-42a537ce-ac1d-11e7-99a3-bf6e0f54ca0d.png)

The issues still to be addressed:

-   [ ] who should be the new project owner, I'm happy to resume administrative duties indefinitely, but I'm also happy for someone else to take over again

-   [ ] licensing: do we act like pirates? can we assume the license was changed in error? do we honour the license on those commits and try to drop them?

**Comment by jokeyrhyme** [comment]
I forgot to mention it before, but I'd very much like to proceed with as much consensus as we can muster, I don't want to be a dictator benevolent or otherwise
I'd also like to recognise and acknowledge all contributors, and not just programmers (translators, people answering questions, documentation writers, triaging issues, etc)

Once we've made a decision with regard to the current license, I'd humbly suggest (if others agree) that we can work in the following areas that are currently blocked:

-   [ ] deciding on a new license: I'd suggest resuming the BSD-2-Clause (a.k.a. Simplified BSD), or maybe switching to MIT or Apache 2.0

-   [ ] deciding on a home for the code: keep it in WowAce Subversion? switch to WowAce git? move it to GitHub?

-   [ ] merging in the latest contributions that have yet to be officially published

-   [ ] publishing a new version to Curse

-   [ ] evangelising that the addon is not dead :)

**Comment by gOOvER** [comment]
i would use Github for the code. Curse can pull it from github. I think its easier when Community want to help

But Repro must changed

**Comment by gOOvER** [comment]
done:

see: https://github.com/eTzmNcbkrng/SavedInstances/pull/56

Basic chjanges for using Github and wowace. Now only changes at Github and WoWace needed

**Comment by Torhal** [comment]
Why are you hard-embedding libraries? That's the whole point of the pkgmeta file.

**Comment by jokeyrhyme** [comment]
@gOOvER I love your enthusiasm, but I would suggest that we resolve the license question before jumping 100% into the technical concerns
you and others are free to disagree, of course :D

**Comment by gOOvER** [comment]
@jokeyrhyme : Never worked with that Licences. I Release all my Code under GPLv3

@Torhal: See my last changes. Added new pkgmeta.yaml file for using packager from Curse :)

**Comment by Stanzilla** [comment]
I guess I'll be fine with `Contributor`, username is the same. Don't have an opinion about the license and would be fine with WowAce git or GitHub with a WowAce webhook.

@Torhal those were only added because we wanted to keep it easy for people to use the GitHub fork.

Also we will have to take a look at the translations again, I merged a few direct additions to the translation files because it did not look like we would ever go back to using the WowAce translation system, but if we start doing that again, that will have to be cleaned up and merged.

**Comment by jokeyrhyme** [comment]
@Stanzilla sorry, I think I need a WowAce username, for some reason it says "stanzilla" is not found (I thought WowAce had been swallowed by Curse but maybe not completely?) https://www.wowace.com/

**Comment by Stanzilla** [comment]
@jokeyrhyme urm, maybe case sensitivity, try Stanzilla

**Comment by jokeyrhyme** [comment]
@Stanzilla done, I bumped you up to Maintainer, since you mentioned translations and Contributor had too few permissions for that

**Comment by Stanzilla** [comment]
Ah true, thanks!

**Comment by Stanzilla** [comment]
@jokeyrhyme do you want me to handle the transition?

**Comment by jokeyrhyme** [comment]
@Stanzilla if you like, I've bumped you up to Author with full permissions
Ideally I'd like to have a few people with full permissions so this doesn't happen again

**Comment by Stanzilla** [comment]
@jokeyrhyme okay, waiting on the decision of license and repo type (gh/wowace) then.

**Comment by TruthNZ** [comment]
My user opinion: Keep on github - it's easy to access, and easy for others to suggest commits via Pull Requests.

Can we proceed with continuing to use oscarucb's code, assume it's ok unless he shows up and asks us to remove it - and deal with it then? Working on the assumption this addon had been freely available to the community thus far.

**Comment by Stanzilla** [comment]
When we go GitHub + Curseforge then it will create zips that include the libs for us anyway. the only real difference is where the source code is, but that does not matter to users.

**Comment by chancedj** [comment]
I was using this addon before it fell out of date (at curse).  I started over with lockedout to mimic a lot of the features in savedinstances and just about have it feature complete.  I was waiting to get a close to that as i could before offering it, but since this debate is happening now i figured i'd offer it up.

You're more then welcome to come help out over on the addon i've started.  It's been mostly coded by myself from scratch (with a little help thrown in here or there) but I've always tried to be very open that anyone is welcome to come help.

If this ends up not getting resolved, or oscarab comes back and says to stop/refuses help - come by and help with lockedout.

**Comment by gOOvER** [comment]
Any News here?

**Comment by Stanzilla** [comment]
Waiting for @jokeyrhyme :)

**Comment by jokeyrhyme** [comment]
Okay, let's start to wrap up this Issue and extract remaining topics out into their own Issues, starting with: https://github.com/eTzmNcbkrng/SavedInstances/issues/61

**Comment by jokeyrhyme** [comment]
Given that the project originally had "do whatever you want" BSD licensing, and this isn't a life-and-death thing, let's go with "it is better to ask for forgiveness than to ask for permission". We'll continue with SavedInstances as-is, and if we later have problems we can drop the offending commits.

Some may want to start looking at @chancedj 's LockedOut addon in the meantime, others may want to hold this option in reserve, that's a personal choice

**Comment by Stanzilla** [comment]
I created a repo on Wowace now and left the license as is https://www.wowace.com/projects/saved_instances/source

**Comment by SacredDuckwhale** [comment]
I'm not sure about the exact terms of the original license, but those that I know (and have looked up again) do have a few, very minor requirements - see ["Free" BSD License](https://opensource.org/licenses/BSD-2-Clause) and ["New" BSD License](https://opensource.org/licenses/BSD-3-Clause) for those that are the go-to choices for most people.

Now, I am not a lawyer but to me, the requirement to keep that notice no matter what other changes are being made implies that you can't just go and change the license at will, even for your own contributions to the project, as the conditions are no longer met once you change the license and remove the file (with the obvious exception of the copyright holder, i.e. @jokeyrhyme, who can do as they please). That's how the GPL handles it, too - do what you want *but don't remove this/change the license*, pretty much.

This means the burden of proof would be on the former maintainer to even assert that his changing the license was legal, and not on you to prove it was not (as that is the reasonable expectation you do get when reading the BSD licenses).

**TL; DR: I agree with the decision to keep using the changes. Sue me! (Ha, ha... please don't)**

PS: As for the maintainers, I can help out with minor tasks such as issues management or smaller fixes, or localization. I just don't have the time to do much with how busy I always am :)

Maybe you'd want to create some sort of "Volunteers" position that can be used for smaller things, while the "authors" would still be able to maintain control and review stuff. There are plenty of people who are willing to provide patches, after all.

**Comment by jokeyrhyme** [comment]
@SacredDuckwhale I've added you as a Maintainer, let's see how we go :)

**Comment by jokeyrhyme** [comment]
I found a few duplicate issues here and closed them (with links to the issues I didn't close): https://www.wowace.com/projects/saved_instances/issues
Got us down from 50+ down to 35

**Comment by SacredDuckwhale** [comment]
Should issues be tracked via GitHub or CurseForge, or both? If GitHub, this repo or another? Unfortunately, the CF issues have some.... issues (heh).

Might be worth discussing separately, perhaps?

**Comment by jokeyrhyme** [comment]
@SacredDuckwhale we discuss the code hosting situation over here: https://github.com/eTzmNcbkrng/SavedInstances/issues/63
I'd suggest that we keep the issues close to the code, so we should update the issues link as appropriate

**Comment by SacredDuckwhale** [comment]
Is there still anything to discuss here? The only item on the list posted above I'm not sure about is this:

>  merging in the latest contributions that have yet to be officially published

Are they all merged? Is something missing?

**Comment by Stanzilla** [comment]
Should be all in now, I think. Otherwise the only thing unclear to me is who all has admin roles on the GitHub org

**Comment by jokeyrhyme** [comment]
Looks like we can close this then :)
